contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -177,31 +177,59 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
         }
     }
 
-    function _patchGenesisValidators(address validator) internal {
+    function _increaseGenesisValidatorCollateral(address validator, uint256 amount) internal {
         // add to initial validators avoiding duplicates if it
         // is a genesis validator.
-        bool alreadyValidator;
         uint256 length = s.genesisValidators.length;
-        for (uint256 i; i < length; ) {
+        uint256 i;
+        for (; i < length; ) {
             if (s.genesisValidators[i].addr == validator) {
-                alreadyValidator = true;
                 break;
             }
             unchecked {
                 ++i;
             }
         }
-        if (!alreadyValidator) {
+        // if we iterate through all of them, and found one
+        // condition must hold
+        if (i < length) {
+            // nth-call to `_increaseGenesisValidatorCollateral`
+            s.genesisValidators[i].weight += amount;
+        } else {
+            // otherwise we need to lookup the power from the current power table
+            // and initialize the amount from that
             uint256 collateral = s.validatorSet.validators[validator].currentPower;
```

Comment:
But it is not being used? @drahnr

Reply:
Deleted the line, I don't think it ever made much sense